rpg.py

```diff
@@ -87,80 +87,105 @@ def __init__(self,
                  hp: int,
                  max_hp: int,
                  attack: int,
-                 speed: int):
+                 speed: int,
+                 leveling_rate: int):
         super().__init__(name,
                          hp,
                          max_hp,
                          attack,
                          speed,
-                         team=1,
-                         level=1,
+                         leveling_rate,
                          exp=0,
-                         target_exp=200)
+                         level_exp=200,
+                         team=1,
+                         level=1)
 
     def act(self, players):
         """the act method specific to team 1"""
 
         all_enemy_locations = self.get_all_enemies(players)
         enemies = {character.name:
                    character for character in players if character.team != 1}
-        if all_enemy_locations:
-            if self.is_alive:
-                if all_enemy_locations.count(1):
+        if self.is_alive:
+            if all_enemy_locations:
+                print(all_enemy_locations)
+                print(enemies)
+                if len(all_enemy_locations) == 1:  # if there is only one enemy in the battle:
+                    print('What move will you use 1?')
                     targeted_index = all_enemy_locations[0]
                     targeted_player = players[targeted_index]
-                    damaged_player = self.deal_damage(targeted_player)
-                    players[targeted_index] = damaged_player
-                else:
+                    damage_input = input('>')
+                    if damage_input == '1':
+                        self.deal_damage(targeted_player)
+                        print(f'{self.name} attacked {targeted_player.name} for {self.attack} damage!')
+                    elif damage_input == '2':
+                        self.deal_damage2(targeted_player)
+                        print(f'{self.name} attacked {targeted_player.name} for {self.attack} damage!')
+
+                else:  # if there is more than one enemy in the battle:
```

Comment:
There's a lot of issues with this code. You never validate `damage_input` or `user_input`, so it will go wrong if a user messed up with inputs.